lib/ReactViews/Analytics/RectangleParameterEditor.tsx

```diff
@@ -1,33 +1,44 @@
-"use strict";
-
 import React from "react";
-
-import PropTypes from "prop-types";
-
 import defined from "terriajs-cesium/Source/Core/defined";
-
 import Styles from "./parameter-editors.scss";
-
 import CesiumMath from "terriajs-cesium/Source/Core/Math";
 import Cartographic from "terriajs-cesium/Source/Core/Cartographic";
 import Ellipsoid from "terriajs-cesium/Source/Core/Ellipsoid";
 import UserDrawing from "../../Models/UserDrawing";
-import { withTranslation } from "react-i18next";
+import RectangleParameter from "../../Models/FunctionParameters/RectangleParameter";
+import { TFunction } from "i18next";
+import { WithTranslation, withTranslation } from "react-i18next";
 import { observer } from "mobx-react";
-import { runInAction } from "mobx";
+import { makeObservable, runInAction } from "mobx";
 import CommonStrata from "../../Models/Definition/CommonStrata";
+import Terria from "../../Models/Terria";
+import ViewState from "../../ReactViewModels/ViewState";
+
+interface IProps extends WithTranslation {
+  previewed: any;
+  parameter: RectangleParameter;
+  viewState: ViewState;
+  t: TFunction;
+}
 
 @observer
-class RectangleParameterEditor extends React.Component {
-  static propTypes = {
-    previewed: PropTypes.object,
-    parameter: PropTypes.object,
-    viewState: PropTypes.object,
-    t: PropTypes.func.isRequired
-  };
+class RectangleParameterEditor extends React.Component<IProps> {
+  constructor(props: IProps) {
+    super(props);
+    makeObservable(this);
+  }
+
+  setValueFromText(e: any) {
+    this.setParameterValueFromText(e, this.props.parameter);
+  }
 
-  setValueFromText(e) {
-    RectangleParameterEditor.setValueFromText(e, this.props.parameter);
+  /**
+   * Triggered when user types value directly into field.
+   * @param e Text that user has entered manually.
+   * @param parameter Parameter to set value on.
+   */
+  setParameterValueFromText(e: any, parameter: RectangleParameter) {
+    parameter.setValue(CommonStrata.user, JSON.parse(e.target.value));
```

Comment:
The `JSON.parse()` was inside array-brackets before my change, but no matter how I shifted type annotations around or trying to cast things I couldn't get that through the type checker. Just dropping the brackets made things pass though, and I can't find any other setValue call that passes an array as the second parameter.